File: 01-数据结构/01-数组/go/LC  装最多水的容器.go
package main

func maxArea(height []int) int {
    l, r := 0, len(height)-1
    max := 0
    for l < r {
        h, w := 0, r - l
        if height[l] <= height[r] {
            h = height[l]
            l++
        }else {
            h = height[r]
            r--
        }
        if w * h > max {
            max = w * h
        }
    }

    return max    
}

File: 01-数据结构/01-数组/c++/LC  装最多水的容器.cpp
#include<vector>

using namespace std;

class Solution {
public:
    int maxArea(vector<int>& height) {
        int max = 0;
        int l = 0;
        int r = height.size() - 1;
        while (l < r) {
            int w = r - l;
            int h = 0;
            if (height[l] < height[r]) {
                h = height[l];
                l++;
            }else {
                h = height[r];
                r--;
            }
            if (h * w > max) {
                max = h * w;
            }
        }

        return max;
    }
};